lib/navigation/preprocess_encoders.cpp

```diff
@@ -34,9 +38,29 @@ core::EncoderData EncodersPreprocessor::detectOutliers(const core::EncoderData e
   -also figure out return type/ what we update and update
   documentation as appropriate
   */
-  return {0, 0, 0, 0};
+ //core::EncoderData encoder_data_copy = std::copy(encoder_data.begin(), encoder_data.end(), std::back_inserter(encoder_data));
+ //core::EncoderData encoder_data_copy = std::copy(encoder_data.begin(), encoder_data.end())
+ core::EncoderData encoder_data_copy;
+ for(int i = 0;i<core::kNumEncoders;i++){
+  encoder_data_copy.at(i) = encoder_data.at(i);
+ }
+ core::Float median = calulate_value(encoder_data_copy);
+ if(encoder_data.size() % 2 == 0){
+    median =  ((encoder_data.at(encoder_data.size()/2) + encoder_data.at((encoder_data.size()/2) + 1))/2.0);
+  }
+  else{
+    median = encoder_data.at(encoder_data.size() + 1)/2.0;
+  }
+ core::Float q1 = encoder_data_copy.at(((encoder_data_copy.size() + 1)/4) -1);
```

Comment:
where are these numbers coming from? if the 4 is number of encoders then `core::kNumEncoders`